astacus/common/rohmustorage.py

```diff
@@ -85,8 +85,9 @@ class RohmuS3StorageConfig(RohmuModel):
 class RohmuAzureStorageConfig(RohmuModel):
     storage_type: Literal[RohmuStorageType.azure]
     account_name: str
-    account_key: str
     bucket_name: str
+    account_key: Optional[str] = None
+    client_config: Optional[Dict[str, Any]] = None
```

Comment:
Please add explicit typing for this too, and also validator which ensures that one or the other is configured. It doesn't seem like client_config is arbitrary dict but instead something with regular structure.

Reply:
Explicit typing here certainly makes sense, thanks for pointing that out.
The validation on the other hand is being implemented in pghoard's AzureTransfer client ([PR #509](https://github.com/aiven/pghoard/pull/509)), which astacus instantiates via the `get_transfer()` method. An `InvalidConfigurationError` is thrown if neither `account_key` nor `client_config` is specified.

Reply:
Yeah, we just rely on our own configuration format being well-formed (instead of funny exception that shows up later).